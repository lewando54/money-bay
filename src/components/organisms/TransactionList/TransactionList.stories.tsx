/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import TransactionList from './TransactionList'

const meta = {
	component: TransactionList,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof TransactionList>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
