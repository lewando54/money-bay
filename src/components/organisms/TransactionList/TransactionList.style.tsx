import { Dimensions, StyleSheet } from 'react-native'

const { width, height } = Dimensions.get('screen')

export default StyleSheet.create({
    transactionsContainer: {
        gap: width * 0.0143
    }
})