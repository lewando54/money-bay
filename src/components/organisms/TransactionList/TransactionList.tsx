import React from 'react'
import TransactionBar, { TTransactionsArray } from '../../molecules/TransactionBar/TransactionBar'
import transactionListStyle from './TransactionList.style'
import Images from 'assets/images'
import { View } from 'react-native'

interface ITransactionListProps {
    transactionsData?: TTransactionsArray
    onDate?: Date
}

export default function TransactionList({
    transactionsData=[
        {
            icon: <Images.transferLineSVG />,
            title: 'Adalyn Roth',
            subTitle: 'Money transfer',
            amount: 140,
            date: new Date('2022-12-02')
        },
        {
            icon: <Images.amazonSVG />,
            title: 'Amazon',
            subTitle: 'Online payments',
            amount: 239.57,
            date: new Date('2022-12-02')
        },
        {
            icon: <Images.payPalSVG />,
            title: 'Paypal',
            subTitle: 'Deposits',
            amount: 700,
            date: new Date('2022-9-10'),
            isExpense: false
        },
    ],
    onDate
}: ITransactionListProps){

    const TransactionListEl = transactionsData.map((item, index) => 
        <TransactionBar
            key={index}
            title={item.title}
            subTitle={item.subTitle}
            amount={item.amount}
            isExpense={item.isExpense}
            icon={item.icon}
        />
    )

    if(onDate){
        const TransactionListFiltered = transactionsData.filter((item) => 
            item.date.toISOString() == onDate.toISOString()
        ).map((item, index) => 
            <TransactionBar
                key={index}
                title={item.title}
                subTitle={item.subTitle}
                amount={item.amount}
                isExpense={item.isExpense}
                icon={item.icon}
            />
        )
        return (
            <View style={transactionListStyle.transactionsContainer}>
                {TransactionListFiltered}
            </View>
        )
    }

    return (
        <View style={transactionListStyle.transactionsContainer}>
            {TransactionListEl}
        </View>
    )
}
