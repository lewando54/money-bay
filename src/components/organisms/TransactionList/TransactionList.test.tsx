import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import TransactionList from './TransactionList'
describe('TransactionList', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<TransactionList />)
    expect(tree).toMatchSnapshot()
  })
})
