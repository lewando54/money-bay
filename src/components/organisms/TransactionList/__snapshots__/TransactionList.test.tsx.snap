// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList should render correctly 1`] = `
<View
  style={
    {
      "gap": 10.725,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 19.95,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 13,
        },
      ]
    }
    testID="universal-container"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F1F5FD",
            "borderRadius": 159.9,
            "height": 79.95,
            "justifyContent": "center",
            "width": 79.95,
          },
          false,
        ]
      }
    >
      <SvgMock />
    </View>
    <View>
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          }
        }
      >
        Adalyn Roth
      </Text>
      <Text
        style={
          {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 12,
          }
        }
      >
        Money transfer
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 5,
          "marginLeft": "auto",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            false,
          ]
        }
      >
        -
      </Text>
      <View>
        <Text
          style={
            [
              {
                "color": "#1B1D4D",
                "fontFamily": "MulishSemiBold",
                "fontSize": 14,
              },
              false,
            ]
          }
        >
          140.00
        </Text>
        <Text
          style={
            [
              {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 12,
              },
              {
                "marginLeft": "auto",
              },
              {
                "display": "none",
              },
            ]
          }
        >
          undefined%
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 19.95,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 13,
        },
      ]
    }
    testID="universal-container"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F1F5FD",
            "borderRadius": 159.9,
            "height": 79.95,
            "justifyContent": "center",
            "width": 79.95,
          },
          false,
        ]
      }
    >
      <SvgMock />
    </View>
    <View>
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          }
        }
      >
        Amazon
      </Text>
      <Text
        style={
          {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 12,
          }
        }
      >
        Online payments
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 5,
          "marginLeft": "auto",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            false,
          ]
        }
      >
        -
      </Text>
      <View>
        <Text
          style={
            [
              {
                "color": "#1B1D4D",
                "fontFamily": "MulishSemiBold",
                "fontSize": 14,
              },
              false,
            ]
          }
        >
          239.57
        </Text>
        <Text
          style={
            [
              {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 12,
              },
              {
                "marginLeft": "auto",
              },
              {
                "display": "none",
              },
            ]
          }
        >
          undefined%
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 19.95,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 13,
        },
      ]
    }
    testID="universal-container"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F1F5FD",
            "borderRadius": 159.9,
            "height": 79.95,
            "justifyContent": "center",
            "width": 79.95,
          },
          false,
        ]
      }
    >
      <SvgMock />
    </View>
    <View>
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          }
        }
      >
        Paypal
      </Text>
      <Text
        style={
          {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 12,
          }
        }
      >
        Deposits
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 5,
          "marginLeft": "auto",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            {
              "color": "#3EB290",
            },
          ]
        }
      >
        +
      </Text>
      <View>
        <Text
          style={
            [
              {
                "color": "#1B1D4D",
                "fontFamily": "MulishSemiBold",
                "fontSize": 14,
              },
              {
                "color": "#3EB290",
              },
            ]
          }
        >
          700.00
        </Text>
        <Text
          style={
            [
              {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 12,
              },
              {
                "marginLeft": "auto",
              },
              {
                "display": "none",
              },
            ]
          }
        >
          undefined%
        </Text>
      </View>
    </View>
  </View>
</View>
`;
